src/Client.ts

```diff
@@ -35,14 +15,33 @@ export type Job = string
  * @experimental
  */
 export default class Client extends EventEmitter {
+	private jobOrderer: JobOrderer
+	private availableDaemons: Set<ClientHttp2Session> = new Set()
+
 	/**
 	 * Create a client whose responsibility is to finish the given jobs.
 	 * It won't start until it knows about some daemons.
 	 *
-	 * @param queue - array of jobs in reverse order
+	 * @param jobs - array of the jobs in any order
 	 */
-	constructor(private readonly queue: Job[]) {
+	constructor(jobs: Job[]) {
 		super()
+		this.jobOrderer = new JobOrderer(jobs)
+	}
+
+	/**
+	 * String representation of the host and port together.
+	 *
+	 * @remarks
+	 * Needed because IPv6 addresses must be bracketed to disambiguate colons.
+	 *
+	 * @param host - hostname or IP address
+	 * @param port - port number
+	 * @returns connection string suitable for a URL
+	 */
+	private static hostAndPort(host: string, port: number): string {
```

Comment:
Actually yeah I don't think that should be in the class, but shouldn't it go in a separate util file? I feel like it's weird just keeping it in `Client.ts` since its such a general function.

Reply:
A separate `Utils.ts` library would be great if it is needed by multiple modules. Until then, there's no harm in it being in `Client.ts` as a free function. It's normal for code to have private implementation details.